bson/_cbsonmodule.c

```diff
@@ -179,6 +188,46 @@ static long long millis_from_datetime(PyObject* datetime) {
     return millis;
 }
 
+/* Extended-range datetime, returns a DatetimeMS object with millis */
+static PyObject* datetime_ms_from_millis(PyObject* self, long long millis){
+    // Allocate a new DatetimeMS object.
+    struct module_state *state = GETSTATE(self);
+
+    PyObject* dt;
+    PyObject* ll_millis;
+
+    if (!(ll_millis = PyLong_FromLongLong(millis))){
+        return NULL;
+    }
+    dt = PyObject_CallFunctionObjArgs(state->DatetimeMS, ll_millis, NULL);
+    Py_DECREF(ll_millis);
+    return dt;
+}
+
+/* Extended-range datetime, takes a DatetimeMS object and extracts the long long value. */
+static long long millis_from_datetime_ms(PyObject* dt){
+    PyObject* ll_millis;
+    long long millis;
+
+    if (!(ll_millis = PyNumber_Long(dt))){
+        if (PyErr_Occurred()) {
+            PyErr_SetString(PyExc_TypeError,
+                            "MongoDB datetimes must be int types.");
```

Comment:
We can remove this PyErr_SetString call completely so that the PyNumber_Long error propagates directly to the user.